src/net/net.c

```diff
@@ -23,39 +23,6 @@
 #define DEBUG_LEVEL 5
 #include <re_dbg.h>
 
-
-/**
- * Get the IP address of the host
- *
- * @param af  Address Family
- * @param ip  Returned IP address
- *
- * @return 0 if success, otherwise errorcode
- */
-int net_hostaddr(int af, struct sa *ip)
-{
-	char hostname[256];
-	struct in_addr in;
-	struct hostent *he;
-
-	if (-1 == gethostname(hostname, sizeof(hostname)))
-		return errno;
-
-	he = gethostbyname(hostname);
-	if (!he)
-		return ENOENT;
-
-	if (af != he->h_addrtype)
-		return EAFNOSUPPORT;
-
-	/* Get the first entry */
-	memcpy(&in, he->h_addr_list[0], sizeof(in));
-	sa_set_in(ip, ntohl(in.s_addr), 0);
-
-	return 0;
-}
-
-
```

Comment:
this is a publicly exported function, and the prototype is defined in `re_net.h`:

```
int  net_hostaddr(int af, struct sa *ip);
```

after this patch, it will only be available for win32. all public functions should be available
for all platforms.

if we want to completely remove it from the API, we must also remove the prototype.

the function `gethostbyname` only works with IPv4 and is deprecated on some platforms.
libre should support both IPv4 and IPv6 properly so I guess it is a good thing to remove it...

Reply:
There is no function `gethostbyname` (may be you mixed up with `gethostname`?), in this PR it was replaced by `GetAddrInfo`
Why platform specific code should be placed in a subfolder? Should I create a header file and .c file for one function? In [net.c](https://github.com/creytiv/re/blob/master/src/net/net.c) it placed in same file, but it used only for Win32

Reply:
it is nice if platform specific code is placed in subfolders, it makes the code
more organized and easy to maintain.

these folders are already under `src/net`:

```
bsd
linux
posix
win32
```